engine/classes/Elgg/Http/ResponseFactory.php

```diff
@@ -407,25 +407,25 @@ public function redirect($forward_url = REFERRER, $status_code = ELGG_HTTP_FOUND
 		$forward_url = elgg_normalize_url($forward_url);
 
 		switch ($status_code) {
-			case 'walled_garden';
-			case 'csrf' :
+			case 'walled_garden':
 			case 'admin' :
 			case 'login' :
-				$status_code = ELGG_HTTP_FORBIDDEN;
+				$status_code = ELGG_HTTP_TEMPORARY_REDIRECT;
```

Comment:
This should be See Other as well. E.g. browsers should not re-send POST requests to /login